src/vellum/workflows/nodes/experimental/tool_calling_node/utils.py

```diff
@@ -0,0 +1,134 @@
+                    elif isinstance(values[0], FunctionCallVellumValue):
+                        function_call = values[0].value
+                        if function_call is not None:
+                            self.state.chat_history.append(
+                                ChatMessage(
+                                    role="FUNCTION",
+                                    content=FunctionCallChatMessageContent(
+                                        value=FunctionCallChatMessageContentValue(
+                                            name=function_call.name,
+                                            arguments=function_call.arguments,
+                                            id=function_call.id,
+                                        ),
+                                    ),
+                                )
+                            )
+            yield output
+
+
+def create_tool_router_node(
+    ml_model: str,
+    blocks: List[PromptBlock],
+    functions: List[FunctionDefinition],
+    prompt_inputs: Optional[EntityInputsInterface],
+) -> Type[ToolRouterNode]:
+    Ports = type("Ports", (), {})
+    for function in functions:
+        if function.name is None:
+            # We should not raise an error here since we filter out functions without names
+            raise ValueError("Function name is required")
+
+        function_name = function.name
+        port_condition = LazyReference(
+            lambda: (
+                ToolRouterNode.Outputs.results[0]["type"].equals("FUNCTION_CALL")
+                & ToolRouterNode.Outputs.results[0]["value"]["name"].equals(function_name)
+            )
+        )
+        port = Port.on_if(port_condition)
+        setattr(Ports, function_name, port)
+
+    setattr(Ports, "default", Port.on_else())
+
+    # Add a chat history block to blocks
+    blocks.append(
+        VariablePromptBlock(
+            block_type="VARIABLE",
+            input_variable="chat_history",
+            state=None,
+            cache_config=None,
+        )
+    )
+
+    node = type(
+        "ToolRouterNode",
+        (ToolRouterNode,),
+        {
+            "ml_model": ml_model,
+            "blocks": blocks,
+            "functions": functions,
+            "prompt_inputs": prompt_inputs,
+            "Ports": Ports,
+            "__module__": __name__,
+        },
+    )
+
+    return node
+
+
+def create_function_node(function: FunctionDefinition, function_callable: Callable[..., Any]) -> Type[FunctionNode]:
+    """
+    Create a FunctionNode class for a given function.
+
+    This ensures the callable is properly registered and can be called with the expected arguments.
+    """
+
+    # Create a class-level wrapper that calls the original function
+    def execute_function(self) -> BaseNode.Outputs:
+        outputs = self.state.meta.node_outputs.get(ToolRouterNode.Outputs.text)
```

Comment:
[oos] on the class below, we should be able to do:

`"function_call": ToolRouterNode.Outputs.results[0]`

and then here we can replace this line and the two below with

`arguments = self.function_call.arguments`